pedalboard/io/PythonInputStream.h

```diff
@@ -32,20 +32,35 @@ bool isReadableFileLike(py::object fileLike) {
          py::hasattr(fileLike, "tell") && py::hasattr(fileLike, "seekable");
 }
 
+std::optional<py::buffer> tryConvertingToBuffer(py::object bufferLike) {
+  try {
+    py::buffer b(bufferLike);
+    return {b};
+  } catch (std::exception &e) {
+    // If not already a buffer, can we call getbuffer() on it and do we then get
+    // a buffer?
+    if (py::hasattr(bufferLike, "getbuffer")) {
+      py::object getBufferResult = bufferLike.attr("getbuffer")();
+      try {
+        py::buffer b(getBufferResult);
+        return {b};
+      } catch (std::exception &e) {
+      }
+    }
+
+    return {};
+  }
+}
+
 /**
  * A juce::InputStream subclass that fetches its
  * data from a provided Python file-like object.
  */
 class PythonInputStream : public juce::InputStream, public PythonFileLike {
```

Comment:
Not sure if you intend to inherit from this class again, but if you don't then [you can add the final specifier](https://en.cppreference.com/w/cpp/language/final).